app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.wallpaper
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.GridCells
+import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.wallpapers.Wallpaper
+
+/**
+ * A grid of selectable wallpaper thumbnails.
+ *
+ * @param wallpapers Wallpapers to add to grid
+ * @param selectedWallpaper The currently selected wallpaper
+ * @param onWallpaperSelected Action to take when a new wallpaper is selected.
+ */
+@Composable
+fun WallpaperSettings(
+    wallpapers: List<Wallpaper>,
+    selectedWallpaper: Wallpaper,
+    onWallpaperSelected: (Wallpaper) -> Unit,
```

Comment:
```suggestion
    onWallpaperSelect: (Wallpaper) -> Unit,
```
I was looking around our codebase and it seems like we prefer the the present tense for the lambda function naming for Jetpack Compose.

Reply:
The present tense looks worse to me but makes more sense logically. I'll update these and then we can push from present tense moving forward for consistency. Though I like `onSelectWallpaper` more than I like `onWallpaperSelect`, what do you think?

Reply:
`onSelectWallpaper` is fine. I saw `onDismissRequest` in `DropdownMenu`. So, when in doubt, just look at some Compose internals.